app/code/Magento/CatalogImportExport/Model/Import/Product/LinkProcessor.php

```diff
@@ -0,0 +1,340 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\CatalogImportExport\Model\Import\Product;
+
+use Magento\Catalog\Model\ResourceModel\Product\LinkFactory;
+use Magento\CatalogImportExport\Model\Import\Product;
+use Magento\Framework\App\ResourceConnection;
+use Magento\Framework\DB\Adapter\AdapterInterface;
+use Magento\ImportExport\Model\Import;
+use Magento\ImportExport\Model\ResourceModel\Helper;
+use Magento\ImportExport\Model\ResourceModel\Import\Data;
+use Psr\Log\LoggerInterface;
+
+class LinkProcessor
+{
+    /**
+     * @var \Magento\CatalogImportExport\Model\Import\Product
+     */
+    protected $_entityModel;
+
+    /**
+     * Links attribute name-to-link type ID.
+     *
+     * @var array
+     */
+    protected $_linkNameToId = [];
+
+    /**
+     * @var LinkFactory
+     */
+    protected $_linkFactory;
+
+    /**
+     * @var LoggerInterface
+     */
+    protected $_logger;
+
+    /**
+     * @var Helper
+     */
+    protected $_resourceHelper;
+
+    /**
+     * @var Data
+     */
+    protected $_dataSourceModel;
+
+    /**
+     * @var SkuProcessor
+     */
+    protected $skuProcessor;
+
+    /**
+     * @var AdapterInterface
+     */
+    protected $_connection;
+
+    /**
+     * @var \Magento\Catalog\Model\ResourceModel\Product\Link
+     */
+    protected $_resource;
+
+    public function __construct(
+        ResourceConnection $resourceConnection,
+        LinkFactory $linkFactory,
+        LoggerInterface $logger,
+        Helper $resourceHelper,
+        Data $importData,
+        SkuProcessor $skuProcessor,
+        array $linkNameToId
+    ) {
+        $this->_linkFactory = $linkFactory;
+        $this->_logger = $logger;
+        $this->_resourceHelper = $resourceHelper;
+        $this->_dataSourceModel = $importData;
+        $this->skuProcessor = $skuProcessor;
+
+        // Temporary solution: arrays do not seem to support init_parameter,
+        // so we have to parse the constant by ourselves
+        $this->_linkNameToId = [];
+        foreach($linkNameToId as $key=>$value) {
+            $this->_linkNameToId[$key] = constant($value);
+        }
+
+        $this->_connection = $resourceConnection->getConnection();
+        $this->_resource = $this->_linkFactory->create();
+    }
+
+    public function saveLinks($entityModel, $productEntityLinkField)
```

Comment:
I'd try to extract a couple of private functions out of this method to make it shorter

Reply:
@sivaschenko I extracted already lots and lots out of this, feel free to extract more, I don't have more ideas here ...

Reply:
see https://github.com/magento/magento2/pull/21658#discussion_r268820332